torch_xla/csrc/tensor.cpp

```diff
@@ -700,6 +701,51 @@ XLATensor XLATensor::rsub(const XLATensor& input, const at::Scalar& other,
   return Create(other_xla - alpha_xla * input.GetIrValue(), input.GetDevice());
 }
 
+XLATensor XLATensor::__and__(const XLATensor& input, const at::Scalar& other) {
+  XLA_CHECK(xla::ShapeUtil::ElementIsIntegral(input.shape()))
+      << "Bitwise and is only supported for integer type tensors";
+  ir::NodePtr other_broadcasted_ir = ir::ops::ScalarOp(other, input.shape());
+  return Create(ir::BitwiseAnd(input.GetIrValue(), other_broadcasted_ir),
+                input.GetDevice());
+}
+
+XLATensor XLATensor::__and__(const XLATensor& input, const XLATensor& other) {
+  XLA_CHECK(xla::ShapeUtil::ElementIsIntegral(input.shape()))
```

Comment:
I am debating about the need for those.
They pollute the code, and if we apply the type-check logic, we should be checking a lot of stuff (like not passing in values to floating point functions, etc... -long-list-).

Reply:
This is required to match the PyTorch behavior, we have to explicitly throw if the input tensors are not integer.

Reply:
PT also throws on x.sin() if x is not floating point ;)

Reply:
Which we will likely do as well, but at later time (XLA Lower).

Reply:
Then maybe `sin` should be fixed as well. However, there's much more of a justification to enforce it for bitwise operations than for arithmetic. Bitwise operations on float tensors are almost certainly unintentional.

Reply:
Also, for `sin` we throw as well and our error message is arguably even a bit better than the PyTorch one. For bitwise, this would *succeed* silently, which is wrong.

Reply:
Looking at the TPU lowering for kAnd ... ouch.
Leave them there for now.